static/carousel/js/carousel.js

```diff
@@ -83,7 +121,7 @@
                 });
 
                 $carousel.slick({
-                    slidesToShow: 3,
+                    slidesToShow: config.no_of_videos,
```

Comment:
Did you declared `no_of_videos ` and the other options on `defaults/carousel.json` ?